lib/hl7/datetime_components.rb

```diff
@@ -0,0 +1,33 @@
+require 'active_support/core_ext/time'
+
+module HL7
+  class DatetimeComponents
+    attr_reader :year, :month, :day, :hour, :minute, :second, :offset
+
+    def initialize(year:, month: 1, day: 1, hour: 0, minute: 0, second: 0, offset: 0)
+      @year = year
+      @month = (month || 1)
+      @day = (day || 1)
+      @hour = (hour || 0)
+      @minute = (minute || 0)
+      @second = (second || 0)
+      @offset = (offset || 0)
```

Comment:
Did you consider naming this `offset_seconds`? I was a little surprised at first to see that this is expected to be seconds. When I read "offset" I assume a string like "-0500." Maybe others who use this would have the same confusion?

Reply:
Also a good point. I also think I should add some basic validation on the fields. (E.g., month must be between 1-12.) Unfortunately, I can't do really good validation without doing a _lot_ of work, since the standard library is rather lacking.